app/components/ad-slot.js

```diff
@@ -61,41 +52,41 @@ export default Component.extend(
      * @returns {void}
      */
     didEnterViewport() {
-      const ads = this.get('ads.module');
-      const name = this.name;
-
-      if (this.noAds) {
-        return;
-      }
-
-      if (this.insertOnViewportEnter) {
-        ads.pushSlotToQueue(name);
-      } else if (this.shouldWaitForUapResponse) {
-        ads.waitForUapResponse(
-          () => ads.pushSlotToQueue(this.name),
-          () => {},
-        );
-      }
+      Ads.getLoadedInstance()
+        .then((ads) => {
+          if (this.insertOnViewportEnter) {
+            ads.pushSlotToQueue(this.name);
```

Comment:
`pushSlotToQueue` does not take any argument. Also, this if statement is kind of broken. In this case, when `insertOnViewportEnter` is truthy, method `pushSlotToQueue` does not take any actions (returns void when `insertOnViewportEnter` is truthy). 🙈

Reply:
https://github.com/Wikia/mobile-wiki/pull/1821/files#diff-98564636413dff58d77852057b876c48R75

Reply:
I tried to refactor this part but then got kind of dizzy looking at how convoluted this is. I will straighten this up.

Reply:
My attempt to clean this place up deserves a ticket on its own.
Leaving it as is for now.